[Manta-PC]Update manta-pc deps to polkadot-v0.9.3

Dengjianping opened the pull request:
Closes #67

Related commits:
Polkadot: [branch=release-v0.9.3](https://github.com/paritytech/polkadot/tree/release-v0.9.3)
Cumulus: [branch=polkadot-v0.9.3](https://github.com/paritytech/cumulus/tree/polkadot-v0.9.3)
Substrate: [branch=polkadot-v0.9.3](https://github.com/paritytech/substrate/tree/polkadot-v0.9.3)

Follow this doc for testing: https://github.com/Manta-Network/Manta/tree/manta-pc-polkadot-v0.9.3/docs

Dengjianping committed:
Update manta-pc deps to polkadot-v0.9.3

ghzlatarev commented:
> Related commits:
> Polkadot: branch=release-v0.9.3
> Cumulus: branch=polkadot-v0.9.3
> Substrate: branch=polkadot-v0.9.3
> 
> Follow this doc for testing: https://github.com/Manta-Network/Manta/tree/manta-pc-polkadot-v0.9.3/docs

Can you be a bit more descriptive with this commit message please ?

ghzlatarev commented:
Can we link an issue and label the PR please ?

stechu commented on pallets/collator-selection/src/lib.rs:
```diff
@@ -0,0 +1,419 @@
+//! Collator Selection pallet.
```
A question here, is this code from parity?

Dengjianping replied:
Yes, this pallet is from statemint. I just wanted to figure out how to do staking as collator, now, it has been removed.

stechu commented on runtime/common/src/lib.rs:
```diff
@@ -0,0 +1,72 @@
+	/// slot_duration()`.
+	///
+	/// Change this to adjust the block time.
+	pub const MILLISECS_PER_BLOCK: u64 = 12000;
```
Is 12 sec the new standard now for collator blocks?

Dengjianping replied:
I just followed statemint's configuration, but how much time it will spend still very unstable, even minutes. Now it's 6_000.

ghzlatarev commented on node/src/command.rs:
```diff
@@ -1,19 +1,3 @@
-// Copyright 2019-2021 Parity Technologies (UK) Ltd.
-// This file is part of Cumulus.
-
-// Cumulus is free software: you can redistribute it and/or modify
-// it under the terms of the GNU General Public License as published by
-// the Free Software Foundation, either version 3 of the License, or
-// (at your option) any later version.
-
-// Cumulus is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-// GNU General Public License for more details.
-
-// You should have received a copy of the GNU General Public License
-// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
-
```
We should add our own copyright, no ?

stechu replied:
yeah, I think @zhenfeizhang  set up a header for `pallet-manta-pay`, should move to this repo as well.

ghzlatarev commented on pallets/collator-selection/README.md:
```diff
@@ -0,0 +1 @@
+License: Apache-2.0
```
Hmm was  under the impression everything was gpl 3

Dengjianping replied:
This pallet has been removed.

ghzlatarev commented on pallets/collator-selection/src/benchmarking.rs:
```diff
@@ -0,0 +1,172 @@
+
+const SEED: u32 = 0;
+
+// TODO: remove if this is given in substrate commit.
```
Is this TODO relevant?

Dengjianping replied:
This pallet has been removed.

Dengjianping committed:
Remove unused pallets, simplify manta-pc runtime, rename runtime-common as manta-primitives

ghzlatarev commented:
> For closing #67
> 
> Related commits:
> Polkadot: [branch=release-v0.9.3](https://github.com/paritytech/polkadot/tree/release-v0.9.3)
> Cumulus: [branch=polkadot-v0.9.3](https://github.com/paritytech/cumulus/tree/polkadot-v0.9.3)
> Substrate: [branch=polkadot-v0.9.3](https://github.com/paritytech/substrate/tree/polkadot-v0.9.3)
> 
> Follow this doc for testing: https://github.com/Manta-Network/Manta/tree/manta-pc-polkadot-v0.9.3/docs

You need to use ```closes #67``` instead of ```closing #67``` in order for the issue to get linked... smart github keywords lol. Linked issues on the right side is still empty.

ghzlatarev commented:
Ooh I just realized this branch doesn't have any CI.
@Dengjianping can you update the check,yml file:

```
on:  
  pull_request:    
       branches: [manta, manta-pc]  
   push:   
        branches: [manta, manta-pc]
```

ghzlatarev commented on runtime/primitives/src/constants.rs:
```diff
@@ -0,0 +1,76 @@
+		fn polynomial() -> WeightToFeeCoefficients<Self::Balance> {
+			// in Polkadot, extrinsic base weight (smallest non-zero weight) is mapped to 1/10 CENT:
+			// in Manta Parachain, we map to 1/10 of that, or 1/100 CENT
+            // revisit here to figure out why use this polynomial
```
If this is our own TODO we should log it.
Somehow rust formatter didn't format it as well.

Dengjianping committed:
Add pallet-sudo, testnet genesis config for manta-pc, setup 1B tokens for initializing acctouns

Dengjianping committed:
Fix fmt, clippy warning, CI/CD support

Dengjianping committed:
Update check.yml

Remove release build for saving disk space, which caused No space left on device

ghzlatarev commented on .github/workflows/check.yml:
```diff
@@ -0,0 +1,60 @@
+  # Triggers the workflow on push or pull request events
+  # but only for the manta (default) branch.
+  pull_request:
+    branches: [manta-pc]
+  push:
+    branches: [manta-pc]
+
```
Hmm, we shouldn't disable it for the ```manta``` branch.

EDIT: nvm

Dengjianping replied:
Now, we are having two branches: `manta` and `manta-pc`, both will not be merged in the near future.

Dengjianping committed:
Expand CI cache

stechu commented on runtime/primitives/src/constants.rs:
```diff
@@ -0,0 +1,77 @@
+		type Balance = Balance;
+		fn polynomial() -> WeightToFeeCoefficients<Self::Balance> {
+			// in Polkadot, extrinsic base weight (smallest non-zero weight) is mapped to 1/10 CENT:
+			// in Manta Parachain, we map to 1/10 of that, or 1/100 CENT
```
This kind of assume MA/DOT ratio is 10:1 ?

stechu reviewed (approved):
LGTM in general. We can merge it first and then figure out the test issue.
